ddtrace/tracer/textmap.go

```diff
@@ -261,9 +268,46 @@ func (p *propagator) injectTextMap(spanCtx ddtrace.SpanContext, writer TextMapWr
 	for k, v := range ctx.baggage {
 		writer.Set(p.cfg.BaggagePrefix+k, v)
 	}
+	if p.cfg.MaxTagsHeaderLen <= 0 {
+		return nil
+	}
+	// propagate trace tags
+	sb := p.formatPropagatingTags(ctx)
+	if sb.Len() > 0 {
+		writer.Set(traceTagsHeader, sb.String())
+	}
 	return nil
 }
 
+func (p *propagator) formatPropagatingTags(ctx *spanContext) strings.Builder {
```

Comment:
I think the name is not quite right. This isn't formatting anything per se, it's creating the tags. So maybe `marshalPropagatingTags` or something like that. Also, why return the builder? Why not just `string`?

Please document this function.